contrib/node/src/python/pants/contrib/node/tasks/node_task.py

```diff
@@ -84,6 +99,22 @@ def execute_npm(self, args, workunit_name=None, workunit_labels=None):
                                  workunit_name=workunit_name,
                                  workunit_labels=workunit_labels)
 
+  def execute_yarnpkg(self, args, workunit_name=None, workunit_labels=None):
```

Comment:
The codes will be cleaner if we have a function `execute` to wrap `execute_npm` and `execute_yarnpkg`.  This way, all conditional codes between npm and yarnpkg can be encapsulated within the `execute` method.